### DISCUSSION TITLE
Help：How widget1，widget2，widget3 communicate with each other？

### DISCUSSION THREAD

**Discussion by ChestnutYYY** [OrchardCMS/OrchardCore#11696] [Q&A]
Hi:
  To do:
 
![image](https://user-images.githubusercontent.com/13895891/168197913-a72e1b23-a5ad-4440-b4d4-7a0161414c9b.png)

These three  partd are ContentPart ,Stereotype are 'Widget'

Problem：
1：Input some text in 'Filter Widget1'
2：Select a filter option in 'Persion list Widget2'
3:   How to refresh 'Widget3' when an option is selected?

**Can it be refreshed without reload the whole page?**

**Comment by ns8482e** [comment]
You can archive using JavaScript - If I were you, I  would create 

- `FilterPart` and `Filter` content item with `FilterPart.cshtml` and `filter.js`
- `PersonPart` and and `Person` content item with `PersonPart.cshtml` and `person.js`

and send ajax request from js files based on my trigger criteria  to custom controller to refresh widget 3
